src/main/scala/io/iohk/ethereum/domain/BlockchainReader.scala

```diff
@@ -71,16 +71,13 @@ class BlockchainReader(
   def getReceiptsByHash(blockhash: ByteString): Option[Seq[Receipt]] = receiptStorage.get(blockhash)
 
   /** get the current best stored branch */
-  def getBestBranch(): Branch =
-    getBestBlock()
-      .map { block =>
-        new BestBranch(
-          block.header,
-          blockNumberMappingStorage,
-          this
-        )
-      }
-      .getOrElse(EmptyBranch$)
+  def getBestBranch(): Branch = {
+    val number = getBestBlockNumber()
+    blockNumberMappingStorage
+      .get(number)
+      .map(hash => BestBranch(hash, number))
+      .getOrElse(EmptyBranch)
```

Comment:
a fallback to the second best branch may be more suitable than an empty branch

Reply:
I added a fallback like in `getBestBlock` to be more consistent